unityfiles/Assets/Scripts/Game/Menu/PauseMenu.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class PauseMenu : MonoBehaviour {
+    [SerializeField]
+    private GameObject pauseMenuCanvas;
+
+    [SerializeField]
+    private GameObject confirmDialog;
+
+    private Boolean isPaused = false;
+
+    private Button continueButton;
+    private static readonly int PAUSE = 0;
+    private static readonly float PLAY = 1;
+
+    private void Awake() {
+        CheckForMissingComponents();
+        
+        DeactivateMenu(); 
+    }
+
+    private void Start() {
+        SubscribeToEvents();
+    }
+
+    // Update is called once per frame
+    void Update() {
+        if (Input.GetButtonDown("Cancel")) {
+            if (confirmDialog.activeSelf) {
+                DeactivateConfirmDialog();
+            } else {
+                TogglePause();
+            }
+        }
+    }
+
+    /// <summary>
+    /// Toggles the pause state.
+    /// Shows pause menu if paused
+    /// </summary>
+    private void TogglePause() {
+        isPaused = !isPaused;
+
+        if (isPaused) {
+            PauseGame();
+        } else {
+            ResumeGame();
+        }
+    }
```

Comment:
å blint flippe en bolean er og be om problemer spesielt unødvendig siden du tester på vedien lengere nede

Reply:
Hva tenker du er problematisk med dette her? Det er jo en Toggle funksjon

Reply:
bedre med:
``` if (!isPaused) {
       PauseGame();
       isPaused=true:
        } else {
            ResumeGame();
            isPaused=false:
        }
```

Reply:
Gjorde de endringene du foreslo, men du må gjerne forklare meg hvorfor du mener det er farlig å flippe en boolean blint.